SourceCode/h4/triangle1.cpp

```diff
@@ -0,0 +1,149 @@
+#include <iostream>
+#include <cmath>
+
+class Vector {
+    float x, y, z;
+public:
+    Vector(float x, float y, float z) {
+        this->x = x;
+        this->y = y;
+        this->z = z;
+    }
+
+    float length() const {
+        return sqrt(x * x + y * y + z * z);
+    }
+    void normalize() {
+        float cachedLength = length();
+        x = x / cachedLength;
+        y = y / cachedLength;
+        z = z / cachedLength;
+    }
+
+    float getX() const {
```

Comment:
Since you have getters and setters for all the members you could use a struct directly.

Reply:
That's a good point. Will add.